src/main/java/org/jenkinsci/plugins/mesos/MesosCloud.java

```diff
@@ -70,13 +85,36 @@ public MesosCloud(String name, String mesosUrl, String jenkinsUrl)
    *
    * @return A future reference to the launched node.
    */
-  private CompletableFuture<Node> startAgent() {
-    // mesos.enqueueAgent().thenCompose(mesosSlave -> mesosSlave.waitUntilOnlineAsync());
-    throw new NotImplementedException();
-  }
-
   @Override
   public boolean canProvision(Label label) {
-    throw new NotImplementedException();
+    // TODO: implement executor limits
+    return true;
+  }
+
+  public MesosApi getMesosClient() {
+    return this.mesos;
+  }
+
+  /**
+   * Start a Jenkins agent.jar on Mesos.
+   *
+   * <p>Provide a callback for Jenkins to start a Node.
+   *
+   * @return A future reference to the launched node.
+   */
+  public Future<Node> startAgent() throws Exception {
+    return mesos
+        .enqueueAgent(this, 0.1, 32)
```

Comment:
do we really mean to hard-code these values?

Reply:
We will follow up with configuration for these.